From f1994bc91688889e4b7ffbca687739023f2a815b Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Fri, 8 Mar 2013 11:11:13 -0800
Subject: iscsiadm: return error when login fails

When login fails because there are too many sessions we should be
returning an error code so that userland is aware that something
is amiss.

Signed-off-by: Hannes Reinecke <hare@suse.com>
Signed-off-by: Lee Duncan <lduncan@suse.com>
---
 usr/session_mgmt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/session_mgmt.c b/usr/session_mgmt.c
index ec1f43a..0b7373f 100644
--- a/usr/session_mgmt.c
+++ b/usr/session_mgmt.c
@@ -178,12 +178,12 @@ int iscsi_login_portal(void *data, struct list_head *list, struct node_rec *rec)
 		goto done;
 	}
 	if (session_count >= rec->session.nr_sessions) {
-		log_debug(1, "%s: %d session%s requested, but %d "
+		log_warning("%s: %d session%s requested, but %d "
 			  "already present.",
 			  rec->iface.name, rec->session.nr_sessions,
 			  rec->session.nr_sessions == 1 ? "" : "s",
 			  session_count);
-		rc = 0;
+		rc = ISCSI_ERR_SESS_EXISTS;
 		goto done;
 	}
 
-- 
1.8.1.4

